ruleset.go

```diff
@@ -127,15 +98,76 @@ type Signature struct {
 	ParamTypes map[string]string
 }
 
-// NewSignature returns the Signature of the given ruleset.
-func NewSignature(rs *Ruleset) *Signature {
-	pt := make(map[string]string)
-	for _, p := range rs.Params() {
-		pt[p.Name] = p.Type
+// NewSignature create
+func NewSignature() *Signature {
```

Comment:
Why don't we just pass in the return type as a constant instead of having all the methods to set it?   I'm not sure what the advantage is?  We always need a return type for it to be a valid signature.